src/ice_spi.c

```diff
@@ -55,42 +60,29 @@ void ice_spi_init(void) {
     gpio_set_dir(ICE_SPI_SCK_PIN, GPIO_IN);
     gpio_set_dir(ICE_SPI_TX_PIN, GPIO_IN);
     gpio_set_dir(ICE_SPI_RX_PIN, GPIO_IN);
+
+    // Initialize SPI, but don't yet assign the pins SPI function so they stay in high impedance mode.
+    spi_init(spi1, 50 * 1000 * 1000);
+
+    // Setup DMA channel and interrupt handler
+    dma_tx = dma_claim_unused_channel(true);
+    dma_rx = dma_claim_unused_channel(true);
+
+    dma_channel_set_irq1_enabled(dma_rx, true);
+    irq_set_exclusive_handler(DMA_IRQ_1, spi_irq_handler);
```

Comment:
`ice_spi.h` is used about everywhere, it justify holding a whole DMA IRQ for it.